src/map.h

```diff
@@ -174,7 +175,7 @@ struct level_cache {
     float seen_cache[MAPSIZE_X][MAPSIZE_Y];
     float camera_cache[MAPSIZE_X][MAPSIZE_Y];
     lit_level visibility_cache[MAPSIZE_X][MAPSIZE_Y];
-    std::bitset<MAPSIZE_X *MAPSIZE_Y> map_memory_seen_cache;
+    std::map<map_memory_layer, std::bitset<MAPSIZE_X *MAPSIZE_Y>> map_memory_seen_cache;
```

Comment:
See #36392 
I don't think we need multiple caches here, fundamentally both render modes render one tile at a time, so I think we can get away with just the one bit per tile.